test/Prompt/DontWrap.C

```diff
@@ -172,7 +172,7 @@ const Test2::Nested::Value::type & Test2::Nested::reference() const {
 Test2::Nested Nest2(80);
 // CHECK: Nested::Nested(160)
 Nest2.simpleAdd(3)
-// CHECK: (Test2::Nested::Value::type) 83
+// CHECK: (int) 83
```

Comment:
I agree that of those choice C. is the better option (even-though it does lose some semantic information compare to A ... which is too verbose.

For the purpose of the PR, I would use C.

For completeness, we could also consider (in the future):

D. (std::vector<int>::iterator) @0x7fad4cf7bbf0
d. (std::vector<int>::value_type) 5
E. (std::vector<int>::iterator) @0x7fad4cf7bbf0
e. (int) 5

where we either somehow find the 'type as spelt' or use name normalization similar to the one use in ROOT (dropping inline namespace and defaulted template parameter).